src/plasma/plasma_manager.cc

```diff
@@ -1352,27 +1409,69 @@ ClientConnection *ClientConnection_init(PlasmaManagerState *state,
   /* Create a new data connection context per client. */
   ClientConnection *conn = new ClientConnection();
   conn->manager_state = state;
-  ClientConnection_finish_request(conn);
   conn->fd = client_sock;
   conn->num_return_objects = 0;
-
   conn->ip_addr_port = client_key;
   state->manager_connections[client_key] = conn;
   return conn;
 }
 
+struct TransferSock {
+
+  static TransferSock get_sock(){
+    int sock;
+    sock = bind_inet_sock(0, false);
+
+    struct sockaddr address;
+    socklen_t addrlen = sizeof(address);
+    getsockname(sock, &address, &addrlen);
+    struct sockaddr_in *address_in = (struct sockaddr_in *) &address;
+    int port = htons(address_in->sin_port);
+
+    CHECK(listen(sock, 128) != -1);
+    return TransferSock(sock, port, address_in);
+  };
+
+  int sock;
+  int port;
+  struct sockaddr_in *address_in;
+
+  TransferSock(int sock, int port, sockaddr_in *address_in){
+    this->sock = sock;
+    this->port = port;
+    this->address_in = address_in;
+  }
+
+};
+
 ClientConnection *ClientConnection_listen(event_loop *loop,
                                           int listener_sock,
                                           void *context,
-                                          int events) {
+                                          int events,
+                                          char conn_type) {
```

Comment:
I'd suggest either separating out the connection types in a separate PR or using a bool like `is_remote` here instead of a character.

Reply:
If you don't separate out the connection types, then could you add a TODO explaining that this connection is shared between local clients and remote plasma managers and should be separated into two connection types?

Reply:
I will do a separate PR for splitting connections, and will make the arg a bool for now.